core/src/main/scala/org/apache/spark/util/Clock.scala

```diff
@@ -99,3 +101,29 @@ private[spark] class SystemClock extends Clock {
     -1
   }
 }
+
+/**
```

Comment:
I meant, we can leverage ManualClock on implementing MockedSystemClock singleton instance. ManualClock contains all the functionality we need in new mechanism of timing, right?

If it is due to the package private, I think it's OK to make it public, with probably pinging some folks in core side.

Reply:
cc. @cloud-fan @jiangxb1987 @Ngone51 
Would we mind if we change the scope of these classes be public? Would we want to ping others to confirm?

Reply:
do you mean `private[spark]`? that's always OK.

Reply:
Changing `private[spark]` to public. I guess it's OK since they're just utility classes and wouldn't be a user facing API.

Reply:
Unfortunately `org.apache.spark.util` is not a private package. Thought we can explicitly mark this file as internal in `SparkBuild.scala`

Reply:
Wait... I guess sql-kafka module should be able to access the package private `private[spark]`...

@jerrypeng would you mind if I ask whether you've tried this before? Otherwise could you please try and see whether it works or not?

Reply:
I think the private[spark] classes in Spark Core module should be accessible by other higher-level packages...

Reply:
Yes. Unless the package private issue exists, I don't see the reason to have this here.

The main reason I want to "isolate" the singleton and the backed class is, once we expose this as utility class, "other tests" in "other modules" can co-use this class, which is unexpected. We will need to narrow the scope as small as possible to avoid any side-effect.
(I know we only expose get and advance so it's hard to break unless concurrent run of test/suites, but just a sort of defensive programming.)

Reply:
@jerrypeng Seems to miss this comment thread.